### PR TITLE
Improve documentation style consitency by implementing Markdown lint. Change "Master" to "Management" account

### PR DESCRIPTION
**Pull request by javydekoning**
*Issue #, if available:*
#491 

*Description of changes:*

- Enforce unordered lists to use `-`
- Enforce max line-length of 80
  - This applies to code blocks as well.
  - Exception are: 
    - `things-without-spaces` such as `[Link-to-console](https://which-frequently-exceeds-80)`
    - Markdown tables
- Rename `Master Account` to `Management account` as per recent AWS Changes
- Remove "quote" style lists items. Using `>` makes it difficult to make changes to list items as they need a `>` prefix per line. (changed to `-`)
- Enforce code-fences to have a language identifier.
- Change file/path emphasis from "*somefile.txt*" to `somefile.txt`
- Enforce ordered lists to be `one_or_ordered`. Which means:

```md
1.
1.
1.
1.
```

Or

```md
1.
2.
3.
4.
```

Github will render both as incremental.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by StewartW — APPROVED** [review]

File: docs/user-guide.md [hunk 1]
```diff
@@ -312,76 +414,119 @@ pipelines:
...
-    infrastructure/
-      some-infra-pipeline.yml
+|-- security/
+|   |-- amazon-guardduty.yml
+|   |-- aws-config.yml
+|-- product-one/
+|   roles/
+|   |-- some-role-pipeline.yml
+|   infrastructure/
+|   |-- some-infra-pipeline.yml
```
Comment (sbkok) [inline docs/user-guide.md:454]: To make it use the same structure as used in the `admin-guide.md` file with lists:

```suggestion
├───security/
|   ├───amazon-guardduty.yml
|   └───aws-config.yml
└───product-one/
    ├───roles/
    |   └───some-role-pipeline.yml
    └───infrastructure/
        └───some-infra-pipeline.yml
```

File: docs/user-guide.md [hunk 2]
```diff
@@ -312,76 +414,119 @@ pipelines:
...
 ```
 
 ### Repositories
 
-Source entities for pipelines can consist of AWS CodeCommit Repositories, Amazon S3 Buckets or GitHub Repositories. Repositories are attached to pipelines in a 1:1 relationship, however, you can choose to clone or bring other repositories into your code during the build phase of your pipeline. You should define a suitable [buildspec](#buildspec) that matches your desired outcome and is applicable to the type of resource you are deploying.
+Source entities for pipelines can consist of AWS CodeCommit Repositories, Amazon
+S3 Buckets or GitHub Repositories. Repositories are attached to pipelines in a
+1:1 relationship, however, you can choose to clone or bring other repositories
```
Comment (sbkok) [inline docs/user-guide.md:461]:
```suggestion
1:1 relationship. However, you can choose to clone or bring other repositories
```

File: docs/user-guide.md [hunk 3]
```diff
@@ -312,76 +414,119 @@ pipelines:
...
+into your code during the build phase of your pipeline. You should define a
+suitable [buildspec](#buildspec) that matches your desired outcome and is
+applicable to the type of resource you are deploying.
 
 ### Removing Pipelines
 
-If you decide you no longer require a specific pipeline you can remove it from the deployment_map.yml file and commit those changes back to the *aws-deployment-framework-pipelines* repository *(on the deployment account)* in order for it to be cleaned up. The resources that were created as outputs from this pipeline will **not** be removed by this process.
+If you decide you no longer require a specific pipeline you can remove it from
```
Comment (sbkok) [inline docs/user-guide.md:468]:
```suggestion
If you decide you no longer require a specific pipeline, you can remove it from
```

File: docs/user-guide.md [hunk 4]
```diff
@@ -312,76 +414,119 @@ pipelines:
...
 
-If you are using [AWS CodeBuild](https://aws.amazon.com/codebuild/) as your build phase you will need to specify a [buildspec.yml](https://docs.aws.amazon.com/codebuild/latest/userguide/build-spec-ref.html) file that will live along side your resources in your repository. This file defines how and what AWS CodeBuild will do during certain phases.
+If you are using [AWS CodeBuild](https://aws.amazon.com/codebuild/) as your
+build phase you will need to specify a
+[buildspec.yml](https://docs.aws.amazon.com/codebuild/latest/userguide/build-spec-ref.html)
+file that will live along side your resources in your repository. This file
+defines how and what AWS CodeBuild will do during certain phases.
 
-Let's take a look an example to breakdown how the AWS Deployment Framework uses `buildspec.yml` files to elevate heavy lifting when it comes to deploying CloudFormation templates.
+Let's take a look an example to breakdown how the AWS Deployment Framework uses
```
Comment (sbkok) [inline docs/user-guide.md:484]:
```suggestion
Let's take a look at an example and breakdown how the AWS Deployment Framework uses
```

File: docs/user-guide.md [hunk 5]
```diff
@@ -312,76 +414,119 @@ pipelines:
...
-
-Other packages such as [cfn-lint](https://github.com/awslabs/cfn-python-lint) can be installed in order to validate that our CloudFormation templates are up to standard and do not contain any obvious errors. If you wish to add in any extra packages you can add them to the *requirements.txt* in the `bootstrap_repository` which is brought down into AWS CodeBuild and installed. Otherwise you can add them into any pipelines specific buildspec.yml.
-
-If you wish to hide away the steps that can occur in AWS CodeBuild, you can move the *buildspec.yml* content itself into the pipeline by using the *spec_inline* property in your map files. By doing this, you can remove the option to have a buildspec.yml in the source repository at all. This is a potential way to enforce certain build steps for certain pipeline types.
+In the example we have three steps to our install phase in our build, the
+remaining phases and steps you add are up to you. In the above steps we simply
+bring in the shared modules we will need to run our main function in
+`generate_params.py`. The $S3_BUCKET_NAME variable is available in AWS CodeBuild
```
Comment (sbkok) [inline docs/user-guide.md:508]:
```suggestion
`generate_params.py`. The `$S3_BUCKET_NAME` variable is available in AWS CodeBuild
```

File: docs/user-guide.md [hunk 6]
```diff
@@ -312,76 +414,119 @@ pipelines:
...
+as we pass this in from our initial creation of the that defines the CodeBuild
```
Comment (sbkok) [inline docs/user-guide.md:509]:
```suggestion
as we pass this in from our deployment map that defines the CodeBuild
```

File: docs/user-guide.md [hunk 7]
```diff
@@ -312,76 +414,119 @@ pipelines:
...
+Project. You do not need to change this.
+
+Other packages such as [cfn-lint](https://github.com/awslabs/cfn-python-lint)
+can be installed in order to validate that our CloudFormation templates are up
+to standard and do not contain any obvious errors. If you wish to add in any
+extra packages you can add them to the `requirements.txt` in the
+`bootstrap_repository` which is brought down into AWS CodeBuild and installed.
+Otherwise you can add them into any pipelines specific buildspec.yml.
```
Comment (sbkok) [inline docs/user-guide.md:517]:
```suggestion
If the requirements are project specific, you can add them into the buildspec.yml
defined in the project itself.
```

File: docs/user-guide.md [hunk 8]
```diff
@@ -416,7 +561,18 @@ pipelines:
...
+When you define CloudFormation templates as artifacts to push through a pipeline
+you might want to have a set of parameters associated with the templates. You
+can utilize the `params` folder in your repository to add in parameters as you
+see fit. To avoid having to create a parameter file for each of the stacks you
+wish to deploy to, you can create a parameter file called `global.yml` (or
+`.json`) any parameters defined in this file will be merged into the parameters
+for any specific account parameter file at build time. For example you might
+have a single parameter for a template called `CostCenter` the value of this
+will be the same across every deployment of your application however you might
```
Comment (sbkok) [inline docs/user-guide.md:572]:
```suggestion
will be the same across every deployment of your application. However, you might
```

File: docs/user-guide.md [hunk 9]
```diff
@@ -510,15 +680,26 @@ It is important to keep in mind that each Deployment Provider *(Code Deploy, Clo
...
+*Note:* Currently only Strings type values are supported as parameters to
+CloudFormation templates when deploying via AWS CodePipeline.
 
 ### Serverless Transforms
 
-If the template that is being deployed contains a transform, such as a Serverless Transform it needs to be packaged and uploaded to S3 in every region where it will be deployed. This can be achieved by setting the `CONTAINS_TRANSFORM` environment variable to *True* in your pipeline definition with a deployment map file. Once the environment variable has been set, within your *buildspec.yml* file you will need to use the *package_transform.sh* helper script (`bash adf-build/helpers/package_transform.sh`). This script will package your template to each region and transparently generate a region specific template for the pipeline deploy stages.
+If the template that is being deployed contains a transform, such as a
+Serverless Transform it needs to be packaged and uploaded to S3 in every region
+where it will be deployed. This can be achieved by setting the
+`CONTAINS_TRANSFORM` environment variable to *True* in your pipeline definition
```
Comment (sbkok) [inline docs/user-guide.md:697]:
```suggestion
`CONTAINS_TRANSFORM` environment variable to `True` in your pipeline definition
```

File: docs/user-guide.md [hunk 10]
```diff
@@ -510,15 +680,26 @@ It is important to keep in mind that each Deployment Provider *(Code Deploy, Clo
...
+with a deployment map file. Once the environment variable has been set, within
+your `buildspec.yml` file you will need to use the `package_transform.sh` helper
```
Comment (sbkok) [inline docs/user-guide.md:699]:
```suggestion
with a deployment map file. After defining the environment variable, update the
`buildspec.yml` file to call the `package_transform.sh` helper
```

File: docs/user-guide.md [hunk 11]
```diff
@@ -510,15 +680,26 @@ It is important to keep in mind that each Deployment Provider *(Code Deploy, Clo
...
+script (`bash adf-build/helpers/package_transform.sh`). This script will package
+your template to each region and transparently generate a region specific
+template  for the pipeline deploy stages.
```
Comment (sbkok) [inline docs/user-guide.md:702]:
```suggestion
template for the pipeline deploy stages.
```

File: docs/user-guide.md [hunk 12]
```diff
@@ -552,40 +741,82 @@ Parameters:
...
-
-To highlight an example of how Parameter Injection can work well, think of the following scenario: You have some value that you wish to rotate on a monthly basis. You have some automation in place that updates the value of a Parameter store parameter on a schedule. Each time this pipeline runs it will check for that value and update the resources accordingly, effectively detaching the parameters from the pipeline itself.
-
-There is also the concept of optionally resolving or importing values. This can be achieved by ending the import or resolve function with a **?**. For example, if you want to resolve a value from Parameter Store that might or might not yet exist you can use an optional resolve *(eg resolve:/my/path/to/myMagicKey?)*. If the key *myMagicKey* does not exist in Parameter Store then an empty string will be returned as the value.
+When you use the special keyword **"resolve:"**, the value in the specified
```
Comment (sbkok) [inline docs/user-guide.md:744]:
```suggestion
When you use the special keyword `resolve:`, the value in the specified
```

File: samples/sample-rdk-rules/README.md [hunk 13]
```diff
@@ -41,50 +60,65 @@ Sample pipeline definition looks like below:
...
 ## Prerequisites/ Important bits
+
 - This solution does not setup config or config recorder.
-- When this solution deploys the config-rule to a target account; it expects config is enabled in the target account.
-- Each target account's config role should be able assume by `<account-that-has-the-lambda-function>` to put evaluations into each target account's config. AKA config role in target account(2222222222) should have the lambda-function-account-id(1111111111) as trusted entity as below.
+- When this solution deploys the config-rule to a target account; it expects
+config is enabled in the target account.
+- Each target account's config role should be able assume by
+`<account-that-has-the-lambda-function>` to put evaluations into each target
+account's config. AKA config role in target account(2222222222) should have the
```
Comment (sbkok) [inline samples/sample-rdk-rules/README.md:121]:
```suggestion
account's config. AKA config role in the target account (2222222222) should have the
```

**Review by sbkok — CHANGES_REQUESTED** [review]
Great improvement, thank you.

I took the liberty to review the lines that were changed.
So most of the comments are on the original documentation, not your PR.

Could you add the following string to the `EXECUTABLE_FILES` definition of `src/lambda_codebase/initial_commit/initial_commit.py`?
```py
    "adf-build/shared/helpers/terraform/install_terraform.sh",
```

The other files are out of scope for that initial commit handler or added already :)

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -53,38 +54,37 @@ Some linters can automatically fix findings by running the command below.
...
-
 ## Contributing via Pull Requests
 
 Contributions via pull requests are much appreciated.
 Before sending us a pull request, please ensure that:
 
 1. You are working against the latest source on the *master* branch.
 2. You check existing open, and recently merged, pull requests to make sure
-   someone else hasn't addressed the problem already.
+someone else hasn't addressed the problem already.
```
[the remark this answers is not in the document]
Comment (javydekoning) [reply CONTRIBUTING.md:64]: Agreed! I'll do that.

**Comment by javydekoning** [comment]
@sbkok I parsed all your comments. Should be much better now. Linter is not catching the list indents yet:

**GOOD**:

```md
1. this is some wrapped
   text
```

**BAD**:

```md
1. this is some wrapped
text
```

I'll submit a feature request for that.

File: docs/user-guide.md [hunk 15]
```diff
@@ -552,40 +741,82 @@ Parameters:
...
+following scenario: You have some value that you wish to rotate on a monthly
+basis. You have some automation in place that updates the value of a Parameter
+store parameter on a schedule. Each time this pipeline runs it will check for
+that value and update the resources accordingly, effectively detaching the
+parameters from the pipeline itself.
+
+There is also the concept of optionally resolving or importing values. This can
+be achieved by ending the import or resolve function with a **?**. For example,
+if you want to resolve a value from Parameter Store that might or might not yet
+exist you can use an optional resolve (e.g. `resolve:/my/path/to/myMagicKey?)`.
```
[the remark this answers is not in the document]
Comment (sbkok) [reply docs/user-guide.md:767]: I think you might have missed this one, @javydekoning.

**Review by sbkok — CHANGES_REQUESTED** [review]
Much appreciated! Thanks for processing the changes.
Only one minor change to go and you would have my approval! :rocket:

File: docs/user-guide.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (javydekoning) [reply docs/user-guide.md]: I did miss it indeed. Fixed now.

**Review by sbkok — APPROVED** [review]
Great improvements to our code base, thank you for adding these!!

**Review by StewartW — APPROVED** [review]

**Pull request merged**